sidekiq/queue.go

```diff
@@ -94,28 +94,75 @@ func NewQueue(client redis.UniversalClient) Queue {
 	// https://github.com/mperham/sidekiq/blob/455e9d56f46f0299eaf3b761596207e15f906a39/lib/sidekiq/fetch.rb#L37
 	// This improves OSS version of sidekiq and will not lose sidekiq jobs.
 	dequeueScript := redis.NewScript(`
-	local sidekiq_ns = ARGV[1]
-	local sidekiq_queue = ARGV[2]
-
-	local queue_key = table.concat({"queue", sidekiq_queue}, ":")
-	if sidekiq_ns ~= "" then
-		queue_key = table.concat({sidekiq_ns, queue_key}, ":")
-	end
+	local queue_ns = ARGV[1]
+	local queue_id = ARGV[2]
 
+	local queue_key = table.concat({queue_ns, queue_id}, ":")
 	return redis.call("lindex", queue_key, -1)
 	`)
 
 	ackScript := redis.NewScript(`
+	local queue_ns = ARGV[1]
+	local queue_id = ARGV[2]
+	local jobm = ARGV[3]
+
+	local queue_key = table.concat({queue_ns, queue_id}, ":")
+	return redis.call("lrem", queue_key, -1, jobm)
+	`)
+
+	dequeueStartScript := redis.NewScript(`
 	local sidekiq_ns = ARGV[1]
 	local sidekiq_queue = ARGV[2]
-	local jobm = ARGV[3]
+	local queue_ns = ARGV[3]
+	local queue_id = ARGV[4]
+	local at = tonumber(ARGV[5])
+	local expire_in_sec = tonumber(ARGV[6])
 
 	local queue_key = table.concat({"queue", sidekiq_queue}, ":")
 	if sidekiq_ns ~= "" then
 		queue_key = table.concat({sidekiq_ns, queue_key}, ":")
```

Comment:
micro improvement:  can avoid doing this work until we call "exists" -- 2 early returns before this variable is used

Reply:
fixed